Bots/DotNet/Skills/CodeFirst/EchoSkillBot/Startup.cs

```diff
@@ -31,23 +33,35 @@ public void ConfigureServices(IServiceCollection services)
         {
             services.AddControllers().AddNewtonsoftJson();
 
-            // Configure credentials
-            services.AddSingleton<ICredentialProvider, ConfigurationCredentialProvider>();
+            var configCredentialProvider = new ConfigurationCredentialProvider(Configuration);
 
-            // Register AuthConfiguration to enable custom claim validation.
-            services.AddSingleton(sp => new AuthenticationConfiguration { ClaimsValidator = new Authentication.AllowedCallersClaimsValidator(sp.GetService<IConfiguration>()) });
+            services.AddSingleton(sp => BotFrameworkAuthenticationFactory.Create(
+                    new ConfigurationChannelProvider(Configuration).ChannelService ?? string.Empty, 
+                    true,
+                    AuthenticationConstants.ToChannelFromBotLoginUrl,
+                    AuthenticationConstants.ToChannelFromBotOAuthScope,
+                    AuthenticationConstants.ToBotFromChannelTokenIssuer,
+                    AuthenticationConstants.OAuthUrl,
+                    AuthenticationConstants.ToBotFromChannelOpenIdMetadataUrl,
+                    AuthenticationConstants.ToBotFromEmulatorOpenIdMetadataUrl,
+                    CallerIdConstants.PublicAzureChannel,
+                    new PasswordServiceClientCredentialFactory(
+                        configCredentialProvider.AppId,
+                        configCredentialProvider.Password,
+                        null,
+                        null),
+                    new AuthenticationConfiguration
+                    {
+                        ClaimsValidator = new AllowedCallersClaimsValidator(new List<string>(Configuration.GetSection("AllowedCallers").Get<string[]>()))
+                    },
+                    null,
+                    null));
```

Comment:
Replace this entire section by:
```suggestion
            var callersSection = Configuration.GetSection("AllowedCallers");
            var allowedCallers = callersSection.Get<string[]>();
            services.AddSingleton(p => new AuthenticationConfiguration
            {
                ClaimsValidator = new AllowedCallersClaimsValidator(allowedCallers)
            });
```